src/AspNetCoreModuleV2/CommonLib/AppOfflineApplication.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#include "AppOfflineApplication.h"
+
+#include <experimental/filesystem>
+#include "HandleWrapper.h"
+
+HRESULT AppOfflineApplication::CreateHandler(IHttpContext* pHttpContext, IREQUEST_HANDLER** pRequestHandler)
+{
+    try
+    {
+        *pRequestHandler = new AppOfflineHandler(pHttpContext, m_strAppOfflineContent);
+    }
+    CATCH_RETURN();
+
+    return S_OK;
+}
+
+HRESULT AppOfflineApplication::OnAppOfflineFound()
+{
+    LARGE_INTEGER   li = {};
+
+    HandleWrapper<InvalidHandleTraits> handle = CreateFile(m_appOfflineLocation.c_str(),
+                        GENERIC_READ,
+                        FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
+                        nullptr,
+                        OPEN_EXISTING,
+                        FILE_ATTRIBUTE_NORMAL,
+                        nullptr);
+
+    RETURN_LAST_ERROR_IF(handle == INVALID_HANDLE_VALUE);
+
+    RETURN_LAST_ERROR_IF (!GetFileSizeEx(handle, &li));
+
+    if (li.HighPart != 0)
+    {
+        // > 4gb file size not supported
+        // todo: log a warning at event log
+        return TRUE;
```

Comment:
Any reason to return `TRUE` instead of `S_OK` here?  Is the goal to ignore the app_offline content if the file is > 4 GB, but still take the app offline?  If so, why limit at 4GB as opposed to another arbitrary size?

Reply:
It should be S_OK. 

> Is the goal to ignore the app_offline content if the file is > 4 GB, but still take the app offline?  

Yes

>  If so, why limit at 4GB as opposed to another arbitrary size

This is the code we inherited from ANCM V1. I think it's just an arbitrary number they picked at that time.